pkg/subctl/cmd/join.go

```diff
@@ -182,22 +182,34 @@ func joinSubmarinerCluster(config *rest.Config, subctlData *datafile.SubctlData)
 	clusterCIDR, err = getPodCIDR(clusterCIDR, networkDetails)
 	exitOnError("Error determining the pod CIDR", err)
 
+	brokerAdminConfig, err := subctlData.GetBrokerAdministratorConfig()
+	exitOnError("Error retrieving broker admin config", err)
+	brokerAdminClientset, err := kubernetes.NewForConfig(brokerAdminConfig)
+	exitOnError("Error retrieving broker admin connection", err)
+	brokerNamespace := string(subctlData.ClientToken.Data["namespace"])
+
 	status.Start("Discovering multi cluster details")
-	globalNetworks, err := globalnet.GetGlobalNetworks(subctlData)
-	exitOnError("Error getting Global network details", err)
+	globalnetInfo, err := globalnet.GetGlobalNetworks(brokerAdminClientset, brokerNamespace)
```

Comment:
This entire logic can be moved to `globalnet` and have single function call from `join.go` 
This will keep join code neat and move complexity to globalnet. Will be easier to add retry logic in globalnet than clutter in here.

Reply:
Let me move it to an API